lib/note-content-editor.tsx

```diff
@@ -152,16 +154,51 @@ export default class NoteContentEditor extends Component {
   };
 
   componentDidUpdate(prevProps) {
-    const { content, filter, noteId, spellCheckEnabled } = this.props;
+    const {
+      content,
+      detectLanguage,
+      filter,
+      noteId,
+      spellCheckEnabled,
+    } = this.props;
+
     const { editorState } = this.state;
 
-    // To immediately reflect the changes to the spell check setting,
-    // we must remount the Editor and force update. The remount is
-    // done by changing the `key` prop on the Editor.
-    // https://stackoverflow.com/questions/35792275/
-    if (spellCheckEnabled !== prevProps.spellCheckEnabled) {
-      this.editorKey += 1;
-      this.forceUpdate();
+    const updateLanguage = () => {
+      const minimumContentLength = 10;
+      if (!detectLanguage || content.text.length < minimumContentLength) {
+        window.spellCheckHandler.switchLanguage(navigator.language);
+        this.setState({ lang: undefined });
+      } else {
+        // Auto-detect the note content language to switch spellchecker
+        window.spellCheckHandler.provideHintText(content.text).then(() => {
+          // Use the auto-detected language to set a `lang` attribute on the
+          // note, which helps Chromium in Electron pick an appropriate font
+          this.setState({
+            lang: window.spellCheckHandler.currentSpellcheckerLanguage,
+          });
+        });
+      }
+    };
+
+    // Only relevant in Electron
+    if (window.spellCheckHandler) {
+      // To immediately reflect the changes to the spell check setting,
+      // we must remount the Editor and force update. The remount is
+      // done by changing the `key` prop on the Editor.
+      // https://stackoverflow.com/questions/35792275/
+      if (spellCheckEnabled !== prevProps.spellCheckEnabled) {
+        this.editorKey += 1;
+        this.forceUpdate();
+        updateLanguage();
```

Comment:
could we instead incorporate the language into the `key` itself so that we don't need to track an increasing number and force a `re-render`?

```ts
getEditorKey = () => {
  const { spellCheckEnabled } = this.props;
  const { lang } = this.state;

  return `${ spellCheckEnabled ? 'spellcheck' : 'no_spellcheck' }-${ lang || 'unknown_lang' }`;
}

render() {
  …
    key={ this.editorKey() }
  …
}
```

With this I believe we should be able to get rid of the force-update cycles and allow the prop-changes to trigger the editor update naturally.

Reply:
replaced in ef9fcb7